packages/phone-number-privacy/signer/test/integration/legacypnp.test.ts

```diff
@@ -404,214 +454,546 @@ describe('legacyPNP', () => {
-  //   // TODO: preserve this test
-  //   it('invalid address returns 400', (done) => {
-  //     const mockRequestData = {
-  //       ...validRequest,
-  //       account: 'd31509C31d654056A45185ECb6',
-  //     }
-
-  //     request(app).post('/getBlindedMessagePartialSig').send(mockRequestData).expect(400, done)
-  //   })
-
-  //   // TODO: preserve this test
-  //   it('invalid hashedPhoneNumber returns 400', (done) => {
-  //     const mockRequestData = {
-  //       ...validRequest,
-  //       hashedPhoneNumber: '+1234567890',
-  //     }
-
-  //     request(app).post('/getBlindedMessagePartialSig').send(mockRequestData).expect(400, done)
-  //   })
-
-  //   // TODO: preserve this test
-  //   it('invalid blinded phone number returns 400', (done) => {
-  //     const mockRequestData = {
-  //       ...validRequest,
-  //       blindedQueryPhoneNumber: '1234567890',
-  //     }
-
-  //     request(app).post('/getBlindedMessagePartialSig').send(mockRequestData).expect(400, done)
-  //   })
-  // })
+  describe(`${SignerEndpoint.LEGACY_PNP_SIGN}`, () => {
+    describe('quota calculation logic', () => {
+      const runLegacyPnpSignQuotaTestCase = async (testCase: legacyPnpQuotaTestCase) => {
+        await prepMocks(
+          testCase.account,
+          testCase.performedQueryCount,
+          testCase.transactionCount,
+          testCase.isVerified,
+          testCase.balanceCUSD,
+          testCase.balanceCEUR,
+          testCase.balanceCELO
+        )
+
+        const req = getLegacyPnpSignRequest(
+          testCase.account,
+          BLINDED_PHONE_NUMBER,
+          AuthenticationMethod.WALLET_KEY,
+          testCase.identifier
+        )
+        const authorization = getPnpRequestAuthorization(req, PRIVATE_KEY1)
+        const res = await sendRequest(req, authorization, SignerEndpoint.LEGACY_PNP_SIGN)
+
+        const { expectedPerformedQueryCount, expectedTotalQuota } = testCase
+        const shouldSucceed = expectedPerformedQueryCount < expectedTotalQuota
+
+        if (shouldSucceed) {
+          expect(res.status).toBe(200)
+          expect(res.body).toMatchObject<SignMessageResponseSuccess>({
+            success: true,
+            version: expectedVersion,
+            signature: expectedSignature,
+            performedQueryCount: expectedPerformedQueryCount + 1, // incremented for signature request
+            totalQuota: expectedTotalQuota,
+            blockNumber: testBlockNumber,
+            warnings: [],
+          })
+        } else {
+          expect(res.status).toBe(403)
+          expect(res.body).toMatchObject<SignMessageResponseFailure>({
+            success: false,
+            version: expectedVersion,
+            performedQueryCount: expectedPerformedQueryCount,
+            totalQuota: expectedTotalQuota,
+            blockNumber: testBlockNumber,
+            error: WarningMessage.EXCEEDED_QUOTA,
+          })
+        }
+      }
+
+      quotaCalculationTestCases.forEach((testCase) => {
```

Comment:
I've added some tests to make sure we're covering all 4 of these cases in the "endpoint functionality" tests, but would prefer to keep running the "quota calculation logic" test cases against both the quota and signing endpoints since it will ensure we are using the exact same rate limit on both endpoiunts and we already have it working that way. Personally, it doesn't bother me to have some redundancy and duplication in the tests. Let me know what you think!